### ISSUE TITLE
Automate creation of plugin data directories

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#2248]
### Issue description
An exception will be thrown if attempting to `getConfig()` when the plugin's data directory has not been created.

This stems from the problem that plugins have to create their data directories manually, which seems stupid to me. Nearly every plugin I've seen has some variation of this code on `onEnable()`:

```php
	public function onEnable() : void{
		if(!is_dir($this->getDataFolder())){
			@mkdir($this->getDataFolder());
		}
		//... whatever else
	}
```
This is bad, generates boilerplate and is fragile. It's easy to forget, resulting in astonishing behaviour.

The same could be said of `saveDefaultConfig()`.

I am thinking that either
a) the server should _always_ pre-create the plugin's data-dir
b) the data dir should be created on the fly (how?)

### Steps to reproduce the issue
Use the following code:
```php
public function onEnable() : void{
    $someValue = $this->getConfig()->get("someValue");
}
```

### OS and versions

* PocketMine-MP: https://github.com/pmmp/PocketMine-MP/commit/30ad3a1705081d7f00ba74fc94963e17249fb7a7
* PHP: 7.2.6
* Server OS: windows
* Game version: irrelevant

**Comment by Frago9876543210** [comment]
And what about subdirectories?

**Comment by dktapps** [comment]
Subdirectories are a different story. Nothing in PocketMine-MP expects there to be subdirectories by default. However, there are lots of things which expect the main directory to exist even though the responsibility for creating it is offloaded to the plugin. This sucks.

**Comment by dktapps** [comment]
The crash with `getConfig()` has been fixed in release 3.0.5. This issue remains as an RFC.

**Comment by dktapps** [comment]
Fixed by 40c28f4d26fd9c0105645171fb3dd5f804f28105.
